nemo_skills/inference/model/__init__.py

```diff
@@ -61,12 +64,17 @@ def get_code_execution_model(server_type, code_execution=None, sandbox=None, **k
     return CodeExecutionWrapper(model=model, sandbox=sandbox, config=code_execution_config)
 
 
-def get_online_genselect_model(server_type, online_genselect_config=None, **kwargs):
+def get_online_genselect_model(model=None, online_genselect_config=None, **kwargs):
     """A helper function to create OnlineGenSelect model."""
-    model = get_model(server_type=server_type, **kwargs)
-    if online_genselect_config is None:
-        online_genselect_config = OnlineGenSelectConfig()
-    return OnlineGenSelectWrapper(model=model, cfg=online_genselect_config)
+    if isinstance(model, str):
+        model = get_model(model=model, **kwargs)
+    return OnlineGenSelectWrapper(model=model, cfg=online_genselect_config or OnlineGenSelectConfig())
+
+
+def get_tool_calling_model(model=None, sandbox=None, **kwargs):
```

Comment:
```suggestion
def get_tool_calling_model(model, sandbox, **kwargs):
```
both are not really optional